FreeRTOS/FreeRTOS-Kernel
# Add support for RV32E core for GCC compiler and add support for FPU (simple and double precision) for GCC compiler

add support for RV32E core for GCC compiler
add support for FPU (simple and double precision) for GCC compiler
Description
Change vPortSetupTimerInterrupt in order to support 64bits RISC-V and modify mtime writing on 32bits RISC-V
Change portASM.s in order to support RV32E (16 registers instead of 32 registers on RV32I)
Change portASM.s in order to save/restore FPU registers on simple/double precision

Test Steps
tested on ARTY board with SIFIVE standard core E31, U54, S54 plus custom core with FPU (simple/double precision)
use "example-freertos-blinky" from freedom-e-sdk from SIFIVE (This change have not been yet committed into our public release, do not hesitate to communicate with me if you want to test it).

Related Issue
NONE

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
